client/azure_client_set.go

```diff
@@ -63,117 +65,185 @@ func NewAzureClientSet(config AzureClientSetConfig) (*AzureClientSet, error) {
+	}
+	virtualNetworkGatewayConnectionsClient, err := newVirtualNetworkGatewayConnectionsClient(c)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	virtualNetworkGatewaysClient, err := newVirtualNetworkGatewaysClient(c)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	virtualMachineScaleSetVMsClient, err := newVirtualMachineScaleSetVMsClient(c)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	virtualMachineScaleSetsClient, err := newVirtualMachineScaleSetsClient(c)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	vnetPeeringClient, err := newVnetPeeringClient(c)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
 	clientSet := &AzureClientSet{
-		DeploymentsClient:                      newDeploymentsClient(c),
-		DNSRecordSetsClient:                    newDNSRecordSetsClient(c),
-		DNSZonesClient:                         newDNSZonesClient(c),
-		GroupsClient:                           newGroupsClient(c),
-		InterfacesClient:                       newInterfacesClient(c),
-		StorageAccountsClient:                  newStorageAccountsClient(c),
-		UsageClient:                            newUsageClient(c),
-		VirtualNetworkClient:                   newVirtualNetworkClient(c),
-		VirtualNetworkGatewayConnectionsClient: newVirtualNetworkGatewayConnectionsClient(c),
-		VirtualNetworkGatewaysClient:           newVirtualNetworkGatewaysClient(c),
-		VirtualMachineScaleSetVMsClient:        newVirtualMachineScaleSetVMsClient(c),
-		VirtualMachineScaleSetsClient:          newVirtualMachineScaleSetsClient(c),
-		VnetPeeringClient:                      newVnetPeeringClient(c),
+		DeploymentsClient:                      deploymentsClient,
+		DNSRecordSetsClient:                    dnsRecordSetsClient,
+		DNSZonesClient:                         dnsZonesClient,
+		GroupsClient:                           groupsClient,
+		InterfacesClient:                       interfacesClient,
+		StorageAccountsClient:                  storageAccountsClient,
+		UsageClient:                            usageClient,
+		VirtualNetworkClient:                   virtualNetworkClient,
+		VirtualNetworkGatewayConnectionsClient: virtualNetworkGatewayConnectionsClient,
+		VirtualNetworkGatewaysClient:           virtualNetworkGatewaysClient,
+		VirtualMachineScaleSetVMsClient:        virtualMachineScaleSetVMsClient,
+		VirtualMachineScaleSetsClient:          virtualMachineScaleSetsClient,
+		VnetPeeringClient:                      vnetPeeringClient,
 	}
 
 	return clientSet, nil
 }
 
-func newDeploymentsClient(config *clientConfig) *resources.DeploymentsClient {
+func newDeploymentsClient(config *clientConfig) (*resources.DeploymentsClient, error) {
 	c := resources.NewDeploymentsClientWithBaseURI(config.resourceManagerEndpoint, config.subscriptionID)
 	c.Authorizer = autorest.NewBearerAuthorizer(config.servicePrincipalToken)
+	err := c.AddToUserAgent(config.partnerIdUserAgent)
 
-	return &c
+	return &c, err
```

Comment:
```suggestion
	return &c, microerror.Mask(err)
```
As Pawel wrote below, errors should be masked / wrapped. See https://github.com/giantswarm/fmt/blob/master/go/errors.md#wrapped-errors

This is required in every function to produce entry in stack trace.